### PR TITLE
docs: Make scope and naming of Pull Requests clearer

### PR DESCRIPTION
**Pull request by thecodingaviator**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

**Comment by RandellDawson** [comment]
@raisedadead and I have been discussing how best to organize the docs in general.  I am not sure if the README.md file is the best location for this.  In fact, I think it would be better somewhere in [Proposing a Pull Request (PR)](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-setup-freecodecamp-locally.md#proposing-a-pull-request-pr) section.

**Comment by thecodingaviator** [comment]
@RandellDawson It would definitely be better in the Proposing a PR section, but not everyone sets up freeCodeCamp locally and then submits PRs, they may entirely miss it

File: docs/README.md [hunk 1]
```diff
@@ -49,3 +49,19 @@ A quick reference to the commands that you will need when working locally.
 | `npm run test:curriculum` | Run the curriculum test suite. |
 | `npm run test:server` | Run the server test suite. |
 | `npm run commit` | An interactive tool to help you build a good commit message. |
+
+## Naming of Pull Requests
```
Comment (raisedadead) [inline docs/README.md]: We prefer a FAQ like tone usually.

```suggestion
## How to prepare a good Pull Request title:
```

File: docs/README.md [hunk 2]
```diff
@@ -49,3 +49,19 @@ A quick reference to the commands that you will need when working locally.
...
+
+When opening a Pull Request(PR), use the following scope table to decide what to title your PR in the following format:
+`fix/add(scope): PR Title`
```
Comment (raisedadead) [inline docs/README.md]: `add` is incorrect. See https://www.conventionalcommits.org/en/v1.0.0-beta.2/

**Comment by RandellDawson** [comment]
> It would definitely be better in the Proposing a PR section, but not everyone sets up freeCodeCamp locally and then submits PRs, they may entirely miss it

@raisedadead Where would you prefer to put this for now?  When we restructure the docs, we can find a more permanent, but I was thinking the README.md was not the best place right now, but I am OK with whatever you think.

**Comment by Nirajn2311** [comment]
IMO I think it's better if a separate doc was created just for making PRs instead of including it in the setting up FCC local doc, as there could be chances once a use gets FCC setup locally they might not go scroll down till the end of the doc

**Comment by raisedadead** [comment]
We could create a separate doc for creating PRs.

**Comment by thecodingaviator** [comment]
I'm opening a new PR with a separate file now. Had messed somethings up here

**Comment by thecodingaviator** [comment]
@raisedadead I'd be glad to have some inputs from you on the scopes for which I haven't been able to write a description

File: docs/README.md [hunk 3]
```diff
@@ -34,18 +34,3 @@ This directory contains all of the documentation on contributing to freeCodeCamp
 
 <a href="/docs/how-to-work-on-guide-articles.md">1. Style guide for creating guide articles.</a><br>
 <a href="/docs/how-to-work-on-coding-challenges.md">2. Style guide for creating coding challenges.</a>
-
```
Comment (RandellDawson) [inline docs/README.md]: What do you plan to do with lines 37-51 which you removed from the README.md?

**Review by raisedadead — APPROVED** [review]
Ignore this. I hit the wrong button

**Review by raisedadead — CHANGES_REQUESTED** [review]
The thing about scopes, is we (as devs) should choose the best that cascades to the entire PR. The scope can vary for each PR.

for example, I personally will choose:

- pages: If all I am changing is pages. Adding removing, updating. Even though I might change other stuff in the PR, as long as the "scope" or the impact are is under pages I am good.
- buttons: If I am doing some microscopic change on one button component for example.
- settings: If I am changing a button that applies only to the settings component.

It's not trivial to explain in docs, but the scope is selected based on the areas that you are targeting and then moving up so that it cascades over the commit or the PR.

When in doubt just leave the scope.

I do not think it's wise to be opinionated on this.

Hence no document can cover all the scopes.

But for your convenience, just limit it to the subdirectories that we have on the root of this repo. Fair and simple.

File: docs/README.md (hunk 3 above)
Comment (thecodingaviator) [reply docs/README.md]: They were removed mistakenly, re-added

File: docs/how-to-open-a-pull-request.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - Fill in the details as they seem fit you. This information will be reviewed and decide whether or not, your pull request is going to be accepted.
+
+    - If the PR is meant to fix an existing bug/issue then, at the end of
+      your PR's description, append the keyword `closes` and #xxxx (where xxxx
+      is the issue number). Example: `closes #1337`. This tells GitHub to
+      automatically close the existing issue, if the PR is accepted and merged.
+
+5. Indicate if you have tested on a local copy of the site or not.
+
+    This is very important when you are making changes that are not copy editing markdown files. For example, changes to CSS or JavaScript code, etc.
```
Comment (RandellDawson) [inline docs/how-to-open-a-pull-request.md]: Not everyone would know what `copy editing` means, so see my suggestion below.  My suggestion can probably still be simplified a bit more though.

```suggestion
    This is very important when you are making changes that are not just making edits to text content such as a Guide article verbiage. Examples of changes needing local testing would include JavaScript, CSS, or HTML which could change the functionality or layout of a page.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Looks like a great start.  See my suggestion.

**Comment by thecodingaviator** [comment]
@raisedadead So we can have it for like curriculum, guide, client, etc. ?

**Comment by thecodingaviator** [comment]
PR is now open for review. @raisedadead I've tried my best to set them as what I could think of them

File: docs/how-to-open-a-pull-request.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# How to open a Pull Request
+
+## How to prepare a good Pull Request title:
+
+When opening a Pull Request(PR), use the following scope table to decide what to title your PR in the following format:
+`fix/feat/chore/refactor/docs/perf (scope): PR Title`
+
+An example is `fix(learn): Fixed tests for the do...while loop challenge`.
+
+| scope | documentation |
```
Comment (RandellDawson) [inline docs/how-to-open-a-pull-request.md]:
```suggestion
| Scope | Documentation |
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@thecodingaviator I think the content looks good.  I only had a minor change request.

File: docs/how-to-open-a-pull-request.md (hunk 5 above)
Comment (thecodingaviator) [reply docs/how-to-open-a-pull-request.md]: Good catch, changes made

**Comment by RandellDawson** [comment]
@raisedadead I am good with the changes now.  If you are fine, go ahead and merge.

**Review by RandellDawson — APPROVED** [review]

File: docs/how-to-open-a-pull-request.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+When opening a Pull Request(PR), use the following scope table to decide what to title your PR in the following format:
+`fix/feat/chore/refactor/docs/perf (scope): PR Title`
+
+An example is `fix(learn): Fixed tests for the do...while loop challenge`.
+
+| Scope | Documentation |
+|---|---|
+| `learn`,`curriculum` | For Pull Requests making changes to the curriculum challenges. |
+| `client` | For Pull Requests making changing to the project settings. |
```
Comment (raisedadead) [inline docs/how-to-open-a-pull-request.md]:
```suggestion
| `client` | For Pull Requests making changes to client platform logic or user interface |
```

File: docs/how-to-open-a-pull-request.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+| `guide` | For Pull Requests which make changes to the guide. |
+| `ui` | For Pull Requests which modify the User Interface of any page. |
```
Comment (raisedadead) [inline docs/how-to-open-a-pull-request.md]:
```suggestion

```

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by thecodingaviator** [comment]
@raisedadead Changes made

File: docs/README.md [hunk 8]
```diff
@@ -48,4 +48,4 @@ A quick reference to the commands that you will need when working locally.
 | `npm run test:client` | Run the client test suite. |
 | `npm run test:curriculum` | Run the curriculum test suite. |
 | `npm run test:server` | Run the server test suite. |
-| `npm run commit` | An interactive tool to help you build a good commit message. |
+| `npm run commit` | An interactive tool to help you build a good commit message. |
```
Comment (raisedadead) [inline docs/README.md]: Looks like nothing is being changed in this file. You should remove this from this commit.

```
git checkout upstream/master -- docs/README.md
```

Comment (thecodingaviator) [reply docs/README.md]: Done

**Review by raisedadead — APPROVED** [review]

**Pull request merged**